File: chronik/chronik-cpp/util/hash.cpp
// Copyright (c) 2022 The Bitcoin developers
// Distributed under the MIT software license, see the accompanying
// file COPYING or http://www.opensource.org/licenses/mit-license.php.

#include <algorithm>
#include <chronik-cpp/util/hash.h>
#include <uint256.h>

namespace chronik::util {

std::array<uint8_t, 32> HashToArray(const uint256 &hash) {
    std::array<uint8_t, 32> array;
    std::copy_n(hash.begin(), 32, array.begin());
    return array;
}

uint256 ArrayToHash(const std::array<uint8_t, 32> &array) {
    uint256 hash;
    std::copy_n(array.begin(), 32, hash.begin());
    return hash;
}

} // namespace chronik::util

File: test/functional/test_framework/hash.py
# Copyright (c) 2024 The Bitcoin developers
# Distributed under the MIT software license, see the accompanying
# file COPYING or http://www.opensource.org/licenses/mit-license.php.
"""
Cryptographic hash functions for the test framework
"""

from .messages import sha256
from .ripemd160 import ripemd160


def hash160(s: bytes) -> bytes:
    return ripemd160(sha256(s))


def hex_be_sha256(data: bytes) -> str:
    return sha256(data)[::-1].hex()
